lint-staged pre-commit hook self-blocks on unrelated bin/render-preview.js error

## Symptom

The Husky pre-commit hook runs `wp-scripts lint-js --fix` via lint-staged when any `src/**/*.{js,jsx}` file is staged. The lint task is invoked without an argument override, so wp-scripts lints the whole project including `bin/render-preview.js` — which has a pre-existing `import/no-extraneous-dependencies` error:

```
/bin/render-preview.js
  64:21  error  'playwright' should be listed in the project's dependencies. Run 'npm i -S playwright' to add it
```

Result: any PR that touches a `src/` JS file is blocked by an error in an unrelated file outside the staged set, and the contributor has to use `--no-verify` (see PR for GH#1584, where the Phase 1 validator commit had to bypass the hook even though the staged files were lint-clean).

## Cause

`package.json` lint-staged config:

```json
"src/**/*.{js,jsx}": ["wp-scripts lint-js --fix"]
```

lint-staged passes the matched file list as positional args to the command — but `wp-scripts lint-js` invokes ESLint with its own glob defaults rather than honouring the passed paths, so it lints far more than just the staged files.

## Fix

Two options:

1. **Switch the lint-staged command** to ESLint directly, e.g.
   `"src/**/*.{js,jsx}": ["eslint --fix"]`. Then lint-staged's positional args go straight to ESLint and only the staged files are linted.

2. **Add `playwright` to devDependencies** so `bin/render-preview.js` no longer errors. Unblocks the hook for everyone immediately but does not address the underlying broad-lint behaviour.

Recommendation: do (1) (correctness) and optionally (2) if `bin/render-preview.js` is meant to be runnable without extra installs.

## Discovered while

Implementing GH#1584 (live block validator). All four hard verification gates passed (PHPUnit 2557/2557, lint:php, lint:js, lint:css, PHPStan, build) but the pre-commit hook still blocked the commit because of `bin/render-preview.js`. Used `--no-verify` to push past per AGENTS.md hook-self-block protocol and filed this issue as the required sibling-bug record.

<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.17.22 plugin for [OpenCode](https://opencode.ai) v1.15.6 with claude-sonnet-4-6 spent 15h 45m and 40 tokens on this as a headless worker.

--- a/package.json
+++ b/package.json
@@ -105,7 +105,7 @@
 	},
 	"lint-staged": {
 		"src/**/*.{js,jsx}": [
-			"wp-scripts lint-js --fix"
+			"eslint --fix"
 		],
 		"src/**/*.css": [
 			"wp-scripts lint-style --fix"

--- a/verify-output.txt
+++ b/verify-output.txt
@@ -17,7 +17,7 @@ ESLint: 8.57.1
 
 ESLint couldn't find the plugin "@wordpress/eslint-plugin".
 
-(The package "@wordpress/eslint-plugin" was not found when loaded as a Node module from the directory "/home/dave/Git/superdav-ai-agent-feature-auto-20260521-235813-gh1702".)
+(The package "@wordpress/eslint-plugin" was not found when loaded as a Node module from the directory "/home/dave/Git/superdav-ai-agent-feature-auto-20260522-010339-gh1600".)
 
 It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:
 

